### PR TITLE
use context instead of redux

### PR DESCRIPTION
**Pull request by AnastasiiaManilnikova**

### CODE REVIEW

File: src/auth/Auth.js [hunk 1]
```diff
@@ -14,9 +13,9 @@ import {
     PENDING,
     FAILURE,
 } from 'utils/requestStatuses'
-import {
-    init,
-} from 'auth/store/actions/init'
+
+import AuthContext from './AuthContext'
+import init from './utils/init'
```
Comment (dimaPortyanka) [inline src/auth/Auth.js:18]: why utils?

File: src/auth/utils/init.js [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+            })
+        })
+        .then((resp) => {
+            setStatus(SUCCESS)
+            setUser({
+                ...resp.user,
+                role: resp.roles,
+                token: keycloak.token,
+            })
+            getMicrosoftToken(setMicrosoftData)
```
Comment (dimaPortyanka) [inline src/auth/utils/init.js:63]: if we havent received microsoft token we can say this is success
multiple set state is also not good better to use useReducer

File: src/main.js [hunk 3]
```diff
@@ -7,5 +7,5 @@ export {
 } from 'auth/hooks/useJWTToken'
 
 export {
-    default as auth,
-} from 'auth/store/reducers'
+    default as useAuthContext,
```
Comment (dimaPortyanka) [inline src/main.js:10]: all that we want to use in other repos should be exported not to make skycell-ag/keycloack/dist...

**Review by dimaPortyanka — CHANGES_REQUESTED** [review]
please fix comments

File: src/auth/Auth.js [hunk 4]
```diff
@@ -13,26 +12,97 @@ import {
     PRISTIN,
     PENDING,
     FAILURE,
+    SUCCESS,
 } from 'utils/requestStatuses'
+import createReducer from 'utils/createReducer'
+import generateAsyncActions from 'utils/generateAsyncActions'
```
Comment (dimaPortyanka) [inline src/auth/Auth.js:18]: can we move it to shared-components?

File: src/auth/Auth.js [hunk 5]
```diff
@@ -13,26 +12,97 @@ import {
...
+
+import AuthContext from './AuthContext'
+import init from './init'
 import {
-    init,
-} from 'auth/store/actions/init'
+    GET_MICROSOFT_TOKEN,
+} from './utils/getMicrosftToken'
```
Comment (dimaPortyanka) [inline src/auth/Auth.js:24]: why utils?

File: src/auth/Auth.js [hunk 6]
```diff
@@ -13,26 +12,97 @@ import {
...
+        dispatch,
+    ] = useReducer(
+        reducer,
+        initState,
+    )
 
     useEffect(() => {
-        dispatch(init())
-    }, [dispatch])
+        init(dispatch)
```
Comment (dimaPortyanka) [inline src/auth/Auth.js:92]: ?

File: package.json [hunk 7]
```diff
@@ -12,19 +12,18 @@
...
     "@skycell-ag/eslint-config": "1.0.3",
+    "cross-env": "7.0.3"
+  },
+  "devDependencies": {
     "eslint-config-airbnb": "18.2.1",
     "eslint-plugin-import": "2.22.1",
     "eslint-plugin-jsx-a11y": "6.4.1",
     "eslint-plugin-react": "7.22.0",
     "eslint-plugin-react-hooks": "4.2.0",
-    "cross-env": "7.0.3",
```
Comment (dimaPortyanka) [inline package.json:27]: should be dev dependency

Comment (AnastasiiaManilnikova) [reply package.json:27]: fixed

File: package.json [hunk 8]
```diff
@@ -1,34 +1,36 @@
...
+    "react": "^17.0.1",
+    "react-dom": "^17.0.1",
+    "react-scripts": "4.0.3",
+    "prop-types": "15.7.2",
+    "@skycell-ag/eslint-config": "1.0.3"
+  },
+  "devDependencies": {
     "react": "^17.0.1",
     "react-dom": "^17.0.1",
     "react-scripts": "4.0.3",
```
Comment (dimaPortyanka) [inline package.json:20]: why dev?

**Review by dimaPortyanka — APPROVED** [review]

**Pull request merged**